portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -183,14 +182,18 @@ public DLFileEntry addFileEntry(
 		String name = String.valueOf(
 			counterLocalService.increment(DLFileEntry.class.getName()));
 
-		String extension = DLAppUtil.getExtension(title, sourceFileName);
+		String extension = FileUtil.getExtension(sourceFileName);
```

Comment:
I think on this point you are presupposing that sourceFileName will never be null.

Reply:
No, if it is null the extension will be empty (exactly the case with external videos, they have no extension).